### ISSUE TITLE
Request for example: Replacing a DOM element

### ISSUE THREAD

**Issue by saurabhnanda** [snoyberg/xml#115]
Can the xml-conduit family of libraries be used to replace an HTML element with another HTML element (say, the output of Lucid) and render the output? I've been struggling to figure something out from the docs, but haven't been able to do so yet.

**Comment by saurabhnanda** [comment]
Just following up on this.

**Comment by k0ral** [comment]
Your question is a bit vague, it would be better if you could provide an example of the desired input/output and how you attempted to implement it.
From what I understand, you might want something like this:

```haskell
-- Both input and output are a stream of XML events
solution :: Conduit Event m Event
solution = parseElements .| replaceElement .| renderElements

-- Implement here what you mean by "HTML element"
data Element

-- Then implement those conduits
parseElements :: Conduit Event m Element
replaceElement :: Conduit Element m Element
renderElements :: Conduit Element m Event
```

**Comment by saurabhnanda** [comment]
Actually, the question is probably simpler than it sounds. I haven't been
able to figure out how to use the DOM  mutation APIs exposed by these
libraries. Did I miss a tutorial/doc?

**Comment by saurabhnanda** [comment]
I've been looking at
https://hackage.haskell.org/package/xml-conduit-1.7.0/docs/Text-XML-Cursor.html
and all the functions seems they're for querying / reading XML elements,
not mutating/writing them.

**Comment by k0ral** [comment]
Indeed, `Text.XML.Cursor` only makes it possible to browse an XML document without mutating it.
I believe you should explore the following solutions instead:
- if the location of the element(s) to replace is known in advance (statically), then zippers or lenses are probably the right tools;
- otherwise, you're bound to scan the whole XML document, in which case a streaming solution is relevant, and you might want to dig my previous answer.

**Comment by kindaro** [comment]
Actually, what's stopping an `Axis` from trying to modify a `Cursor` in a specified way and returning a singleton if successfully done so? It seems to me that a "replace" axis that swaps the current node of a Cursor for a given one is a natural thing to implement.
